### ISSUE TITLE
How to access swagger path/request URL from within handler

### ISSUE THREAD

**Issue by christianklotz** [go-swagger/go-swagger#395]
## Problem statement
I want to create a `Link` header containing information for paginating content. What's the best way to create values based on the swagger path/request URL? 

```go
api.BooksListBooksHandler = books.ListBooksHandlerFunc(func() middleware.Responder {
	var list = []*models.Book{
		&models.Book{Title: "Winnie the Pooh", Summary: "Famous children's book"},
	}

	// How to get the URL based on the request?
	links := []string{
		`<https://api.bookstore.com/v1/?page=2>; rel="next"`,
		`<https://api.bookstore.com/v1/?page=13>; rel="last"`,
	}
	return books.NewListBooksOK().WithLink(strings.Join(links, ",")).WithPayload(list)
})

```

## Swagger specification
```yaml
swagger: "2.0"
info:
  title: Swagger Sample
  description: Sample API Playground.
  version: 1.0.0
basePath: /v1
schemes:
- http
consumes:
- application/vdn.sample.v1+json
produces:
- application/vdn.sample.v1+json

paths:
  /books:
    get:
      summary: List all books
      operationId: listBooks
      tags:
        - books
      responses:
        200:
          headers:
            Link:
              type: string
              example: "<https://api.bookstore.com/books?page=2>; rel=\"next\",
      <https://api.github.com/resource?page=5>; rel=\"last\""
          description: An array of books
          schema:
            type: array
            items:
              $ref: "#/definitions/Book"
        default:
          description: generic error response
          schema:
            $ref: "#/definitions/Error"

definitions:
  # Book
  Book:
    type: object
    required:
      - title
      - summary
    properties:
      title:
        type: string
        example: Winnie the Pooh
      summary:
        type: string
        example: Famous children's book

  # Error
  Error:
    type: object
    readOnly: true
    properties:
      code:
        type: integer
        format: int64
        example: 400
      message:
        type: string
        example: Unexpected error
    required:
      - message

```

**Comment by ontucker** [comment]
I'm interested in this as well, since I'm currently trying to figure out a good way to access the TLS client info from inside a handler func.  I figure I can do some middleware trickery to get at the http.Request but I'm curious if I'm overlooking some more straightforward way to plumb it down into the handler.

**Comment by casualjim** [comment]
this was fixed on 0.6.0

**Comment by Marclev78** [comment]
What was the fix that answers the OP's question?
